wp-includes/class-wp-web-app-manifest.php

```diff
@@ -83,15 +83,51 @@ public function manifest_link_and_meta() {
 				?>
 				<meta name="apple-mobile-web-app-capable" content="yes">
 				<meta name="mobile-web-app-capable" content="yes">
+
 				<?php
 				$icons = isset( $manifest['icons'] ) ? $manifest['icons'] : array();
 				usort( $icons, array( $this, 'sort_icons_callback' ) );
 				$icon = array_shift( $icons );
-				?>
-				<?php if ( ! empty( $icon ) ) : ?>
-					<link rel="apple-touch-startup-image" href="<?php echo esc_url( $icon['src'] ); ?>">
-				<?php endif; ?>
-				<?php
+
+				$images = array();
+				if ( ! empty( $icon ) ) {
+					$images[] = array( 'href' => $icon['src'] );
+				}
+
+				/**
+				 * Filters splash screen images for Safari on iOS.
+				 *
+				 * @param array $images {
+				 *     Array of splash screen images and their attributes.
+				 *
+				 *     @type array ...$0 {
+				 *         Array of splash screen image attributes.
+				 *
+				 *         @type string $href  URL of splash screen image. Required.
+				 *         @type string $media Media query for when the splash screen image should be used.
+				 *     }
+				 * }
+				 */
+				$images = apply_filters( 'apple_touch_startup_image', $images );
+
+				foreach ( $images as $key => $image ) {
+					if ( ! is_array( $image ) ) {
+						continue;
+					}
+
+					$url = isset( $image['href'] ) ? esc_url( $image['href'], array( 'http', 'https' ) ) : '';
+					if ( ! $url ) {
+						continue;
+					}
+
+					$html = sprintf( 'rel="apple-touch-startup-image" href="%s"', $url );
+					if ( isset( $image['media'] ) ) {
+						$html .= sprintf( ' media="%s"', esc_attr( $image['media'] ) );
+					}
+					?>
+					<link <?php echo $html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
+					<?php
```

Comment:
It would seem a bit better to do:

```suggestion
					printf( '<link rel="apple-touch-startup-image" href="%s"', esc_attr( $url ) );
					if ( isset( $image['media'] ) ) {
						printf( ' media="%s"', esc_attr( $image['media'] ) )
					}
					echo '>';
```

In this way the tag markup is all constructed together and there is no need to suppress any phpcs warnings.

Reply:
I based this code on [`wp_resource_hints()`](https://github.com/WordPress/wordpress-develop/blob/710f30eb966091408ebb89c276bec6a8f6f12a1d/src/wp-includes/general-template.php#L3384) (curious how that line doesn't trigger phpcs warnings) and I think separating HTML generation and output looks cleaner, but I will make this change.